liquibase-core/src/main/java/liquibase/snapshot/jvm/ColumnSnapshotGenerator.java

```diff
@@ -200,7 +200,7 @@ protected void addTo(DatabaseObject foundObject, DatabaseSnapshot snapshot) thro
                     Column column = readColumn(row, relation, database);
                     setAutoIncrementDetails(column, database, snapshot);
                     populateValidateNullableIfNeeded(column, metaDataNotNullConst, database);
-                    column.setAttribute(LIQUIBASE_COMPLETE, !column.isNullable());
+                    column.setAttribute(LIQUIBASE_COMPLETE, true);
```

Comment:
I agree that this was very strange logic. I went back to the original (internal to Liquibase) bug ticket that was linked to the PR where this `!column.isNullable()` was introduced. I tried to replicate the reported bug both with and without this change. Both code paths work correctly. My best guess is that at some point, the `validate` attribute was not in the XSD for `addNotNullConstraint` and that somehow made that other code useful. 

In any case, I am no longer worried about this change breaking a bug fixed a very, very long time ago.